### PR TITLE
langchain[patch]: add stop for various non-openai agents

### PR DESCRIPTION
**Pull request by mackong**
* Description: add stop for various non-openai agents.
* Issue: N/A
* Dependencies: N/A

### CODE REVIEW

**Comment by eyurtsev** [comment]
Looks reaasonable

File: libs/langchain/langchain/agents/json_chat/base.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ def create_json_chat_agent(
     tools: Sequence[BaseTool],
     prompt: ChatPromptTemplate,
     stop_sequence: bool = True,
+    stop: Optional[list[str]] = None,
```
[the remark this answers is not in the document]
Comment (mackong) [reply libs/langchain/langchain/agents/json_chat/base.py:19]: Fixed

File: libs/langchain/langchain/agents/json_chat/base.py [hunk 2]
```diff
@@ -15,7 +15,7 @@ def create_json_chat_agent(
     llm: BaseLanguageModel,
     tools: Sequence[BaseTool],
     prompt: ChatPromptTemplate,
-    stop_sequence: bool = True,
+    stop_sequence: Union[bool, List[str]] = True,
```
Comment (eyurtsev) [inline libs/langchain/langchain/agents/json_chat/base.py:18]: Could you confirm you're OK with this change?

I noticed that there was an existing stop_sequence in this API. If so, I thin we should make the other APIs consistent

Comment (mackong) [reply libs/langchain/langchain/agents/json_chat/base.py:18]: LGTM， better than another stop parameter

**Review by eyurtsev — APPROVED** [review]

**Pull request merged**